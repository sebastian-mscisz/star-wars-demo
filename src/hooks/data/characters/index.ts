export * from './useCharacters';
export * from './useCharacterDetails';
