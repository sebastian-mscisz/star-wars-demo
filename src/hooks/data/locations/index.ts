export * from './useLocationDetails';
export * from './useLocations';
