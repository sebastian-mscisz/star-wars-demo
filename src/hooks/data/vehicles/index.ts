export * from './useVehicleDetails';
export * from './useVehicles';
