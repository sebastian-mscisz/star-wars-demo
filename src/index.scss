body {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    padding: 0;
    margin: 0;
    background-image: url('/wallpaper.jpg');
    background-repeat: no-repeat;
    background-color: black;
    width: 100%;
    font-family: sans-serif;
}

#root {
    display: flex;
    flex-wrap: wrap;
    flex-grow: 1;
}
