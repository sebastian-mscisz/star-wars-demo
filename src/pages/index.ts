export * from './character-details-page';
export * from './characters-page';
export * from './location-details-page';
export * from './locations-page';
export * from './master-page';
export * from './not-found-page';
export * from './vehicle-details-page';
export * from './vehicles-page';
