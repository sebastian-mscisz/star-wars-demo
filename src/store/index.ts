export * from './data-selectors';
export * from './slices';
export * from './utility';
export * from './rootReducer';
export * from './store';
