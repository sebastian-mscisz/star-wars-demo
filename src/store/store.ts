import { configureStore, EnhancedStore } from '@reduxjs/toolkit';

import rootReducer from './rootReducer';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const createStore = (): EnhancedStore<any, any> => {
    const store = configureStore({
        reducer: rootReducer,
    });

    return store;
};

export const store = createStore();

export type AppDispatch = typeof store.dispatch;
