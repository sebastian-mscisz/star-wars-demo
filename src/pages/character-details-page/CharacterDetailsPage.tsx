import { CharacterDetails } from 'src/components';
import { useRequiredParams } from 'src/common';

export const CharacterDetailsPage: React.FC = () => {
    const { characterId } = useRequiredParams<{ characterId: string }>();

    return <CharacterDetails characterId={characterId} />;
};
