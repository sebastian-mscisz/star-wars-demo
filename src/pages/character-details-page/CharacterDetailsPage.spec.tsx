import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { CharacterDetailsPage } from './CharacterDetailsPage';
import { screen, render } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import { DataApi } from 'src/xhr';
import {
    characterDetailsPagePath,
    getCharactersEndpoint,
    getLocationsEndpoint,
    getSpeciesEndpoint,
    getVehiclesEndpoint,
} from 'src/common';
import { Provider } from 'react-redux';
import { createStore } from '../../store';
import { TestCharacters, TestSpecies } from 'src/components/characters/test-data';
import { TestVehicles } from 'src/components/vehicles/test-data';
import { TestLocations } from 'src/components/locations/test-data';

describe('CharacterDetailsPage', () => {
    let dataApiMock: MockAdapter;

    beforeEach(() => {
        dataApiMock = new MockAdapter(DataApi);
    });

    it('should render', async () => {
        dataApiMock.onGet(getCharactersEndpoint()).reply(200, TestCharacters);
        dataApiMock.onGet(getVehiclesEndpoint()).reply(200, TestVehicles);
        dataApiMock.onGet(getLocationsEndpoint()).reply(200, TestLocations);
        dataApiMock.onGet(getSpeciesEndpoint()).reply(200, TestSpecies);

        const { asFragment } = render(
            <Provider store={createStore()}>
                <MemoryRouter initialEntries={[`${characterDetailsPagePath('4')}`]}>
                    <Routes>
                        <Route path="/character/:characterId" element={<CharacterDetailsPage />} />
                    </Routes>
                </MemoryRouter>
            </Provider>,
        );

        await screen.findByText('Character4');

        expect(asFragment()).toMatchSnapshot();
    });
});
