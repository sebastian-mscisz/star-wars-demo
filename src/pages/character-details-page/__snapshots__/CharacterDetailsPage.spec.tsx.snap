// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetailsPage should render 1`] = `
<DocumentFragment>
  <div
    class="sw-character-detail"
  >
    <div
      class="sw-character-detail__name"
    >
      <div
        class="sw-character-detail__name__image-wrapper"
      >
        <img
          alt="Star wars character"
          src="./characters/4.jpg"
        />
      </div>
      <h3>
        Character4
      </h3>
    </div>
    <div
      class="sw-character-detail__details"
    >
      <p>
        <strong>
          Origin world:
        </strong>
         
        <a
          class="sw-character-detail__details__link"
          href="/location/4"
        >
          Location4
        </a>
      </p>
      <hr />
      <p>
        <strong>
          Race:
        </strong>
         Species4
      </p>
      <hr />
      <p>
        <strong>
          Used vehicles:
        </strong>
         
        <a
          class="sw-character-detail__details__link"
          href="/vehicle/3"
        >
          Vehicle3
        </a>
      </p>
    </div>
  </div>
  <div
    class="sw-pagination-section sw-pagination-section--center"
    data-testid="pagination-section"
  >
    <div
      class="sw-pagination-button "
      data-testid="left-pagination-button"
    >
      ←
    </div>
    <div
      class="sw-pagination-button "
      data-testid="right-pagination-button"
    >
      →
    </div>
  </div>
</DocumentFragment>
`;
